stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -239,6 +247,462 @@ lazy_static! {
+            tenure_id_consensus_hash
+        )
+    }
+
+    pub fn highest_block_in_tenure(tenure_id_consensus_hash: &ConsensusHash) -> String {
+        format!(
+            "nakamoto::tenures::highest_block_in_tenure::{}",
+            tenure_id_consensus_hash
+        )
+    }
+
+    pub fn coinbase_height(ch: &ConsensusHash) -> String {
+        format!("nakamoto::headers::coinbase_height::{}", ch)
+    }
+
+    pub fn tenure_start_block_id(ch: &ConsensusHash) -> String {
+        format!("nakamoto::headers::tenure_start_block_id::{}", ch)
+    }
+
+    pub fn finished_tenure_consensus_hash(ch: &ConsensusHash) -> String {
+        format!("nakamoto::tenures::finished_tenure_consensus_hash::{}", ch)
+    }
+
+    pub fn parent_tenure_consensus_hash(ch: &ConsensusHash) -> String {
+        format!("nakamoto::tenures::parent_tenure_consensus_hash::{}", ch)
+    }
+
+    pub fn make_block_id_value(id: &StacksBlockId) -> String {
+        format!("{}", id)
+    }
+
+    pub fn make_consensus_hash_value(ch: &ConsensusHash) -> String {
+        format!("{}", ch)
+    }
+
+    pub fn make_u64_value(value: u64) -> String {
+        to_hex(&value.to_be_bytes())
+    }
+
+    pub fn make_bool_value(value: bool) -> String {
+        to_hex(&[if value { 1 } else { 0 }])
+    }
+
+    pub fn make_tenure_id_value(value: &NakamotoTenureId) -> String {
+        format!("{}{}", &value.burn_view_consensus_hash, &value.block_id)
+    }
+
+    pub fn parse_consensus_hash(value: &str) -> Option<ConsensusHash> {
+        ConsensusHash::from_hex(value).ok()
+    }
+
+    pub fn parse_block_id(value: &str) -> Option<StacksBlockId> {
+        StacksBlockId::from_hex(value).ok()
+    }
+
+    pub fn parse_u64(value: &str) -> Option<u64> {
+        let bytes = hex_bytes(value).ok()?;
+        if bytes.len() != 8 {
+            return None;
+        }
+        let mut bytes_u64 = [0u8; 8];
+        bytes_u64[0..8].copy_from_slice(&bytes[0..8]);
+        Some(u64::from_be_bytes(bytes_u64))
+    }
+
+    pub fn parse_bool(value: &str) -> Option<bool> {
+        let bytes = hex_bytes(value).ok()?;
+        if bytes.len() != 1 {
+            return None;
+        }
+        Some(bytes[0] != 0)
+    }
+
+    pub fn parse_tenure_id_value(value: &str) -> Option<NakamotoTenureId> {
+        let bytes = hex_bytes(value).ok()?;
+        if bytes.len() != 52 {
+            // ConsensusHash is 20 bytes
+            // StacksBlockId is 32 bytes
+            return None;
+        }
+        let mut ch_bytes = [0u8; 20];
+        let mut block_id_bytes = [0u8; 32];
+        ch_bytes[0..20].copy_from_slice(&bytes[0..20]);
+        block_id_bytes[0..32].copy_from_slice(&bytes[20..52]);
+
+        let id = NakamotoTenureId {
+            burn_view_consensus_hash: ConsensusHash(ch_bytes),
+            block_id: StacksBlockId(block_id_bytes),
+        };
+        Some(id)
+    }
+}
+
+/// Trait for common MARF getters between StacksDBConn and StacksDBTx
+pub trait StacksHandle {
```

Comment:
This trait seems like it could do a bunch of deduplication by just requiring `get_indexed()` and then providing the default implementation for almost all of these function.

Reply:
Sounds like a good idea